exercise/views.py

```diff
@@ -144,29 +145,40 @@ def post(self, request, *args, **kwargs):
             self.submission_check(True, request)
         )
         if submission_allowed:
-            new_submission = Submission.objects.create_from_post(
-                self.exercise, students, request)
-            if new_submission:
-                page = self.exercise.grade(request, new_submission,
-                    url_name=self.post_url_name)
-                for error in page.errors:
-                    messages.error(request, error)
-
-                # Enroll after succesfull enrollment exercise.
-                if self.exercise.status in (
-                    LearningObject.STATUS.ENROLLMENT,
-                    LearningObject.STATUS.ENROLLMENT_EXTERNAL,
-                ) and new_submission.status == Submission.STATUS.READY:
-                    self.instance.enroll_student(self.request.user)
-
-                # Redirect non AJAX normally to submission page.
-                if not request.is_ajax() and "__r" not in request.GET:
-                    return self.redirect(new_submission.get_absolute_url() +
-                        ("?wait=1" if page.is_wait else ""))
+            try:
+                new_submission = Submission.objects.create_from_post(
+                    self.exercise, students, request)
+            except ValueError as error:
+                messages.error(
+                    request,
+                    format_lazy(
+                       _("The submission could not be saved for some reason. "
+                         "The submission was not registered. error: {}"),
```

Comment:
Typo: error -> Error

Did you try this in practice? Does the formatting look good when the Exception is included in the string? Was there a specific reason for using `format_lazy` since I thought that the normal `.format` method works?

Reply:
It looks fine, no specific reason for lazy_format just blindly following examples. I changed it to .format, which looks fine too.

Reply:
> Was there a specific reason for using format_lazy since I thought that the normal .format method works?

_() returns a lazy string and using `.format` on it retrieves the translation at that spot. If for some reason the code is executed somewhere where language is not yet defined correctly, then the wrong translation code is retrieved. In other words. It's always good to use `format_lazy` with `_`. (aka. `ugettext_lazy`)